draft-ietf-mls-protocol.md

```diff
@@ -655,80 +584,57 @@ public-key encryption to the public key K of the secret value S):
 | pk(B)      |             |
 
 
-### Blank Ratchet Tree Nodes
-
-Nodes in a ratchet tree can have a special value "\_", used to indicate that the
-node should be ignored during path computations. Such nodes are used to replace
-leaves when participants are deleted from the group.
+# Group State
 
-If any node in the copath of a leaf is \_, it should be ignored during the
-computation of the path. For example, the tree consisting of the private
-keys (A, \_, C, D) is constructed as follows for ART:
+Each participant in the group maintains a representation of the
+state of the group:
 
 ~~~~~
-  DH(A, DH(CD))
-   /      \
-  A       DH(CD)
- / \      /  \
-A   _    C    D
+struct GroupState {
+  opaque group_id<0..255>;
+  uint32 size;
+  uint32 epoch;
+  Credential roster<1..2^24-1>;
```

Comment:
Wondering where is __Credential__ defined ?

Reply:
Good catch, I forgot to do that.